anitya/lib/librariesio_consumer.py

```diff
@@ -0,0 +1,217 @@
+
+        This serves two purposes. Firstly, we connect to libraries.io over HTTP
+        so we don't have any authentication. Secondly, we might map the
+        libraries.io project to the wrong Anitya project, so this is a good
+        opportunity to catch and log those problems.
+        """
+        librariesio_version = message['body']['msg']['data']['version']
+        try:
+            anitya_version = get_version(project)
+        except exceptions.AnityaPluginException as e:
+            _log.warning('libraries.io found an update for %r, but we failed with %s',
+                         project, str(e))
+            return False
+
+        # TODO handle things like 'v' prefix. Should be a general version normalizer
+        # used everywhere.
+        # TODO handle when we find a different version than librariesio. Create the project anyway?
+        equivalent_versions = (librariesio_version == anitya_version)
+        if not equivalent_versions:
+            _log.info('libraries.io found %r had a latest version of %s, but Anitya found %s',
+                      project, librariesio_version, anitya_version)
+        return equivalent_versions
+
+    def _fedmsg_to_candidate_project_update(self, message):
+        """
+        Transform a librariesio fedmsg into a potential Anitya project update.
+
+        If the project is unknown to Anitya, a new :class:`Project` is created.
+        If it's an existing project, this will speculatively update the latest
+        version information.
+
+        .. warning::
+            This message is *not* authenticated and its authenticity should be
+            confirmed by checking the upstream source.
+
+        Args:
+            message (dict): The fedmsg received by :meth:`self.consume`
+
+        Returns:
+            Project: the Anitya project that maps to the librariesio project;
+                this may be a new project. Note that this is not committed to
+                the database: that is the responsibility of the caller after
+                validating the update.
+            None: If there is no ecosystem that maps to the librariesio platform
+        """
+        librariesio_msg = message['body']['msg']['data']
+        name = librariesio_msg['name']
+        platform = librariesio_msg['platform']
+        version = librariesio_msg['version']
+        homepage = librariesio_msg['package_manager_url']
+        for ecosystem in ECOSYSTEM_PLUGINS.get_plugins():
+            if ecosystem.librariesio_name == platform:
+                break
+        else:
+            _log.debug('Dropped librariesio update to %s for %s (%s) since it is on the '
+                       'unsupported %s platform', version, name, homepage, platform)
+            return
+
+        session = Session()
+        project = Project.by_name_and_ecosystem(session, name, ecosystem.name)
+        if project is None:
+            # We've found a new project!
+            project = Project(
```

Comment:
A problem with this approach that I've just found is that we're not going to be emitting the correct fedmsgs since those are sent all over the place via ``anitya.log``.

I could re-work this PR to use that approach, but I'm inclined to open a new PR that adds [ORM event listeners](http://docs.sqlalchemy.org/en/latest/orm/events.html) that publish the messages instead.